overlord/snapstate/flags.go

```diff
@@ -50,3 +54,32 @@ func (f Flags) ForSnapSetup() Flags {
 	f.IgnoreValidation = false
 	return f
 }
+
+// EffectiveConfinement return the effective confinement for a given declared
+// confinement and flags like jailmode or devmode.
+func (f Flags) EffectiveConfinement(confinementType snap.ConfinementType) snap.ConfinementType {
+	switch confinementType {
+	case snap.ClassicConfinement:
+		// unchanged, flags don't impact snaps with classic confinement
+	case snap.DevmodeConfinement:
+		if f.JailMode {
+			// jailmode overrides devmode confinement declaration
+			return snap.StrictConfinement
+		}
+		// unchanged, stays devmode
+	case snap.StrictConfinement:
+		// In case both jailmode and devmode are set in flags, look at jailmode
+		// first. This errs on the safer side of being strict.
+		if f.JailMode {
+			// jailmode flag overrides devmode flag
+			return snap.StrictConfinement
+		}
+		if f.DevMode {
```

Comment:
`DevMode` or `Devmode`?  I think the latter is what we have in most cases currently, and also reflects the use of "devmode" and "jailmode" without dashes (single word).